mlflow/utils/annotations.py

```diff
@@ -29,11 +31,44 @@ def _experimental(api: Any, api_type: str):
     return api
 
 
+def developer_stable(func):
+    """
+    The API marked here as `@developer_stable` has certain protections associated with future
+    development work.
+    Classes marked with this decorator implicitly apply this status to all methods contained within
+    them.
+
+    APIs that are annotated with this decorator are guaranteed (except in cases of notes below) to:
+    - maintain backwards compatibility such that earlier versions of any MLflow client, cli, or
+      server will not have issues with any changes being made to them from an interface perspective.
+    - maintain a consistent contract with respect to existing named arguments such that
+      modifications will not alter or remove an existing named argument.
+    - maintain implied or declared types of arguments within its signature.
+    - maintain consistent behavior with respect to return types.
+
+    Note: Should an API marked as `@developer_stable` require a modification for enhanced feature
+      functionality, a deprecation warning will be added to the API well in advance of its
+      modification.
+
+    Note: Should an API marked as `@developer_stable` require patching for any security reason,
+      advanced notice is not guaranteed and the labeling of such API as stable will be ignored
+      for the sake of such a security patch.
```

Comment:
Q: 
In this case, shall we document that the dev-stable API behavior has changed since XX version because of some security fix ?

Reply:
That's a good idea!